usaspending_api/common/query_with_filters.py

```diff
@@ -422,16 +423,47 @@ def generate_elasticsearch_query(cls, filter_values: List[str], query_type: _Que
         nested_path = options.get("nested_path", "")
         def_codes_query = []
         def_code_field = f"{nested_path}{'.' if nested_path else ''}disaster_emergency_fund_code{'s' if query_type != _QueryType.ACCOUNTS else ''}"
-        for v in filter_values:
-            def_codes_query.append(ES_Q("match", **{def_code_field: v}))
-        if query_type == _QueryType.TRANSACTIONS:
-            return ES_Q(
-                "bool",
-                should=def_codes_query,
-                minimum_should_match=1,
-                must=ES_Q("range", action_date={"gte": "2020-04-01"}),
-            )
-        return ES_Q("bool", should=def_codes_query, minimum_should_match=1)
+
+        all_covid_defc = set(
+            DisasterEmergencyFundCode.objects.filter(group_name="covid_19").values_list("code", flat=True)
+        )
+        covid_filters = list(set(filter_values) & all_covid_defc)
+        other_filters = list(set(filter_values) - all_covid_defc)
+
+        covid_queries = [ES_Q("match", **{def_code_field: v}) for v in covid_filters]
+        other_queries = [ES_Q("match", **{def_code_field: v}) for v in other_filters]
+
+        if covid_queries:
+            if query_type == _QueryType.TRANSACTIONS:
+                query = ES_Q(
+                    "bool",
+                    must=ES_Q("range", action_date={"gte": "2020-04-01"}),
+                    should=covid_queries,
+                    minimum_should_match=1,
+                )
+            elif query_type == _QueryType.AWARDS:
```

Comment:
Why add a special case for Awards requiring non-zeroes?

Reply:
Special case already exists, but is in the `_handle_defc_query` function down below. Instead of sharing similar logic in two places around DEFC I combined it into one location.

Reply:
Oh I see that now